server/pbench/bin/index-pbench

```diff
@@ -881,7 +915,109 @@ class ToolData(object):
     def _make_source_individual(self):
         """Read .csv files individually, emitting records for each row and
         column coordinate."""
-        pass
+        for csv in self.files:
+            assert csv['header'][0] == 'timestamp_ms'
+            header = csv['header']
+            handler_rec = csv['handler_rec']
+            reader = csv['reader']
+            first_ts = None
+
+            for row in reader:
+                datum = _dict_const()
+                klass = handler_rec['class']
+                datum[klass] = _dict_const()
+                metric = handler_rec['metric']
+
+                # Some individual handlers have the id in the filename pattern.
+                if 'pattern' in handler_rec:
+                    # Does the pattern match the basename?
+                    match = handler_rec['pattern'].match(csv['basename'])
+                    if match:
+                        try:
+                            # This handler has the id in the filename pattern.
+                            datum[klass][metric] = _dict_const([('id', match.group('id'))])
+                        except IndexError as ie:
+                            # This individual handler does not have an id.
+                            datum[klass][metric] = _dict_const()
+
+                for idx,val in enumerate(row):
+                    # The timestamp column is index zero.
+                    if idx == 0:
+                        time_val = float(val)/1000
+                        utc = datetime.utcfromtimestamp(time_val)
+                        # Convert the first timestamp to a formatted string.
```

Comment:
Is this converting only the first timestamp found or all timestamps found?  Seems like the `first_ts` is really just the "current" timestamp.

Reply:
correct